pkg/trace/api/transports.go

```diff
@@ -0,0 +1,139 @@
+		if rerr != nil {
+			tags := append(m.tags, fmt.Sprintf("error:%s", fmt.Sprintf("%T", rerr)))
+			metrics.Count(fmt.Sprintf("%s.proxy_request_error", m.prefix), 1, tags, 1)
+		}
+	}(time.Now())
+	return m.rt.RoundTrip(req)
+}
+
+// forwardingTransport is an HTTP transport wraps another transport that
+// forwards a request to multiple endpoints. The first target in the targets
+// slice is considered the main endpoint. Only the main endpoints response will
+// be returned to the client. Responses of additional endpoints in the targets
+// slice are dropped. Errors on additional endpoints will be logged.
+type forwardingTransport struct {
+	rt      http.RoundTripper
+	targets []*url.URL
+	keys    []string
+}
+
+// newForwardingTransport creates a new forwardingTransport, wrapping another
+// round tripper with a main endpoint and additional endpoints to forwards the
+// request to.
+func newForwardingTransport(
+	rt http.RoundTripper,
+	mainEndpoint *url.URL,
+	mainEndpointKey string,
+	additionalEndpoints map[string][]string,
+) *forwardingTransport {
+	targets := []*url.URL{mainEndpoint}
+	apiKeys := []string{mainEndpointKey}
+	if additionalEndpoints != nil {
+		for endpoint, keys := range additionalEndpoints {
+			u, err := url.Parse(endpoint)
+			if err != nil {
+				log.Errorf("Error parsing additional intake URL %s: %v", endpoint, err)
+				continue
+			}
+			for _, key := range keys {
+				targets = append(targets, u)
+				apiKeys = append(apiKeys, strings.TrimSpace(key))
+			}
+		}
+	}
+	return &forwardingTransport{rt, targets, apiKeys}
+}
+
+// RoundTrip makes an HTTP round trip forwarding one request to multiple
+// additional endpoints.
+func (m *forwardingTransport) RoundTrip(req *http.Request) (rres *http.Response, rerr error) {
+	setTarget := func(r *http.Request, u *url.URL, apiKey string) {
+		q := r.URL.Query()
+		u.RawQuery = q.Encode()
+		r.Host = u.Host
+		r.URL = u
+		r.Header.Set("DD-API-KEY", apiKey)
+	}
+	if len(m.targets) == 1 {
+		setTarget(req, m.targets[0], m.keys[0])
+		return m.rt.RoundTrip(req)
+	}
+
+	slurp, err := io.ReadAll(req.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, u := range m.targets {
+		newreq := req.Clone(req.Context())
+		newreq.Body = io.NopCloser(bytes.NewReader(slurp))
+		setTarget(newreq, u, m.keys[i])
+		if i == 0 {
+			// given the way we construct the list of targets the main endpoint
+			// will be the first one called, we return its response and error
+			rres, rerr = m.rt.RoundTrip(newreq)
+			continue
+		}
+
+		if resp, err := m.rt.RoundTrip(newreq); err == nil {
+			// we discard responses for all subsequent requests
+			io.Copy(io.Discard, resp.Body) //nolint:errcheck
+			resp.Body.Close()
+		} else {
+			log.Error(err)
+		}
+	}
+	return rres, rerr
+}
+
+// newMeasuringMultiTransport creates a MultiTransport wrapped in a MeasuringTransport.
+func newMeasuringMultiTransport(
```

Comment:
Nit: we've named it forwarding transport, not MultiTransport (although I don't dislike the name if you want to rename it).